File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        compliments = {}
        for i in range(len(nums)):
            if target - nums[i] in compliments.keys():
                return [compliments.get(target - nums[i]), i]
            compliments[nums[i]] = i

        return []

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int startPointer = 0;
        int endPointer = numbers.length - 1;

        while (startPointer < endPointer) {
            if (numbers[startPointer] + numbers[endPointer] == target) {
                return new int[]{startPointer + 1, endPointer + 1};
            } else if (numbers[startPointer] + numbers[endPointer] > target) {
                endPointer--;
            } else {
                startPointer++;
            }
        }
        return new int[]{};
    }
}
